nrefocus/iface/base.py

```diff
@@ -136,8 +136,13 @@ def autofocus(self, interval, metric="average gradient", minimizer="lmfit",
             else:
                 raise ValueError(f"Unexpected valud for `roi`: '{roi}'")
         elif roi is None:
-            # Use all the data
-            roi = slice(None, None)
+            if metric == 'spectrum':
```

Comment:
Since there might be more spectrum-like metrics, I would propose to instead change the code in `mt_spectrum.py` instead:

```python
if roi not in [slice(None), None]:
    raise ...
```

Reply:
Would this work with `roi=slice(None)`? I feel like it would throw the same error. I can check it

Reply:
With the current implementation, only None will be passed to the metric. The above might be needed if we want to allow slices to be given as `roi`.